gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java

```diff
@@ -168,27 +167,12 @@ private void submit(Job job) throws ClassNotFoundException, IOException, Interru
 
   private void moveTmpPathToOutputPath() throws IOException {
     LOG.info(String.format("Moving %s to %s", this.tmpPath, this.outputPath));
+    this.fs.delete(this.outputPath, true);
```

Comment:
The old logic deletes outputPath even earlier (before MR job is run), which is why the old logic didn't work if inputPath = outputPath.
The outputPath has to be deleted at some stage so that the compacted data can be moved to the outputPath.